### PR TITLE
chore(glean): update utm.* descriptions & data sensitivity and add event.reason

### PR DESCRIPTION
**Pull request by chenba**

### CODE REVIEW

**Comment by chenba** [comment]
@akkomar

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml [hunk 1]
```diff
@@ -169,10 +203,15 @@ utm:
...
     data_sensitivity:
-      - web_activity
+      - interaction
 
   source:
-    description: UTM source
+    description: |
+      The source from where the user started.  For example, if the user clicked on a
+      link on the Firefox accounts web site, this value could be 'fx-website'.  The
+      value has a max length of 128 characters with the alphanumeric characters, _
```
Comment (badboy) [inline packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:212]: Drive-by comment: string metrics are limited to 100 characters ([docs](https://mozilla.github.io/glean/book/reference/metrics/string.html)). Everything about that will be truncated. If this is a problem for this data collection we need to know and figure out if we should raise the limit or change the collection.

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml [hunk 2]
```diff
@@ -20,6 +20,23 @@ event:
     data_sensitivity:
       - web_activity
```
Comment (akkomar) [inline packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:21]:
```suggestion
      - interaction
```

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml [hunk 3]
```diff
@@ -185,10 +224,16 @@ utm:
       - https://bugzilla.mozilla.org/show_bug.cgi?id=1830504
     expires: never
     data_sensitivity:
-      - web_activity
+      - interaction
 
   term:
-    description: UTM term
+    description: |
+      This metric is similar to the UTM source above; it is used in the Firefox
```
Comment (akkomar) [inline packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:231]:
```suggestion
      This metric is similar to the `utm.source`; it is used in the Firefox
```

Nit: remove `above`, it would be confusing when someone's reading this description in Glean Dictionary or Data Catalog.

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml (hunk 1 above)
Comment (chenba) [reply packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:212]: Thanks for pointing that out.  Even though our validation allows up to 128 chars I've never seen any value remotely close to that length, so I think we should be fine.

File: packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml (hunk 3 above)
Comment (chenba) [reply packages/fxa-shared/metrics/glean/fxa-ui-metrics.yaml:231]: > ... confusing when someone's reading this description in Glean Dictionary or Data Catalog.

Great point.  👍

**Comment by chenba** [comment]
We've got a data-review+ so I'll rebase, squash, and merge.  Thank you for the feedback.

**Review by dschom — APPROVED** [review]
Not super familiar with these configs, but they seem good to me.
